src/exporter.h

```diff
@@ -0,0 +1,191 @@
+#ifndef EXPORTER_H_
+#define EXPORTER_H_
+
+#include <array>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "backend.h"
+#include "daosoa.h"
+#include "default_force.h"
+#include "inline_vector.h"
+#include "math_util.h"
+#include "param.h"
+
+
+using std::fstream;
+using namespace std;
+
+namespace bdm {
+
+class Exporter {
+ public:
+  template <typename daosoa>
+  void ToFile(daosoa* cells, string filename) const {
+    // This function exports the cell positions into a file,
+    // where each line contains the 3D position of a cell in square brackets.
+    const size_t n_vectors = cells->vectors();
+
+    ofstream outfile;
+    outfile.open("../demo/" + filename);
+
+    double currPosX, currPosY, currPosZ;
+
+    for (size_t j = 0; j < VcBackend::kVecLen; j++) {
+      for (size_t i = 0; i < n_vectors; i++) {
```

Comment:
please reverse these two loops
for efficiency first iterate over all vectors and than all scalars inside the vector 
also it contains a bug: the last vector might not be full
->
```
  for (size_t i = 0; i < n_vectors; i++) {
    for (size_t j = 0; j < (*cells)[i].Size(); j++)  {
    ...